src/Data/Vec/Relation/Unary/AllPairs.agda

```diff
@@ -0,0 +1,82 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Vectors where every pair of elements are related (symmetrically)
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary using (Rel)
+
+module Data.Vec.Relation.Unary.AllPairs
+       {a ℓ} {A : Set a} {R : Rel A ℓ} where
+
+open import Data.Vec.Base using (Vec; []; _∷_)
+open import Data.Vec.Relation.Unary.All as All using (All; []; _∷_)
+open import Data.Product as Prod using (_,_; _×_; uncurry; <_,_>)
+open import Function using (id; _∘_)
+open import Level using (_⊔_)
+open import Relation.Binary as B using (Rel; _⇒_)
+open import Relation.Binary.Construct.Intersection renaming (_∩_ to _∩ᵇ_)
+open import Relation.Binary.PropositionalEquality
+open import Relation.Unary as U renaming (_∩_ to _∩ᵘ_) hiding (_⇒_)
+open import Relation.Nullary using (yes; no)
+import Relation.Nullary.Decidable as Dec
+open import Relation.Nullary.Product using (_×-dec_)
+
+------------------------------------------------------------------------
+-- Definition
+
+open import Data.Vec.Relation.Unary.AllPairs.Core public
+
+------------------------------------------------------------------------
+-- Operations
+
+head : ∀ {x n} {xs : Vec A n} → AllPairs R (x ∷ xs) → All (R x) xs
+head (px ∷ pxs) = px
+
+tail : ∀ {x n} {xs : Vec A n} → AllPairs R (x ∷ xs) → AllPairs R xs
+tail (px ∷ pxs) = pxs
+
+uncons : ∀ {x n} {xs : Vec A n} → AllPairs R (x ∷ xs) → All (R x) xs × AllPairs R xs
+uncons = < head , tail >
+
+module _ {q} {S : Rel A q} where
```

Comment:
Probably be better to name the level `s` rather than `q` to match the name of the relation?